pkg/controllermanager/controller/shoot/shoot_conditions_control.go

```diff
@@ -141,3 +134,23 @@ func FilterSeedForShootConditions(obj, oldObj, _ client.Object, _ bool) bool {
 
 	return false
 }
+
+// AddSeedConditionsToShoot adds the conditions from the seed to the shoot by adapting the type of the seed's SystemComponentsHealthy condition so that it is not duplicated.
+func AddSeedConditionsToShoot(seed *gardencorev1beta1.Seed, shoot *gardencorev1beta1.Shoot) []gardencorev1beta1.Condition {
+	seedConditionTypes := []gardencorev1beta1.ConditionType{
+		gardencorev1beta1.SeedBackupBucketsReady,
+		gardencorev1beta1.SeedBootstrapped,
+		gardencorev1beta1.SeedExtensionsReady,
+		gardencorev1beta1.SeedGardenletReady,
+		SeedSystemComponentsHealthyInShoot,
+	}
+	conditions := gardencorev1beta1helper.RemoveConditions(shoot.Status.Conditions, seedConditionTypes...)
+	if seed != nil {
+		seedSystemComponentsCondition := gardencorev1beta1helper.GetOrInitCondition(seed.Status.Conditions, gardencorev1beta1.SeedSystemComponentsHealthy)
+		seedSystemComponentsCondition.Type = SeedSystemComponentsHealthyInShoot
```

Comment:
TBH: I don't like that the type is changed here, because it makes things implicit and confusing.
It would be better to call the condition type `SeedSystemComponentsHealthy` everywhere (seed and shoot).

Reply:
We already had a discussion about this, with @plkokanov  and @timuthy and we decided to this approach to avoid duplication in the conditions names of the Shoot.

Reply:
I think @timebertt means here to directly name the condition "SeedSystemComponentsHealthy" instead of having two names for it, but there was also a comment for the opposite back from when the PR was opened. You can check @rfranzke 's first review comment (not sure how I can provide a link only to it :( ).

Reply:
This was also my preference in the beginning, as I wanted to avoid the potentially complex condition merge. @rfranzke are you ok to use `SeedSystemComponentsHealthy` everywhere?

Reply:
Go ahead, I don't even have the context anymore.

Reply:
> This was also my preference in the beginning, as I wanted to avoid the potentially complex condition merge.

Then let's please call it `SeedSystemComponentsHealthy` everywhere.